packages/host/app/services/matrix-service.ts

```diff
@@ -78,6 +83,10 @@ export default class MatrixService extends Service {
   profile = getMatrixProfile(this, () => this.client.getUserId());
 
   rooms: TrackedMap<string, Promise<RoomField>> = new TrackedMap();
+  messagesToSend: TrackedMap<string, string | undefined> = new TrackedMap();
```

Comment:
I'm guessing that the reason is that we wanted this state to be longer lived than this component. Like if you clicked the send button, closed the chat window, and then reopened it, if the message was still pending you'd be able to see that since the room component had to be recreated when you reopened the chat window.

Reply:
Right, that's my thinking, we need the states to be lived longer than the component.